src/vm/threads.cpp

```diff
@@ -8715,31 +8654,21 @@ OBJECTREF Thread::GetCulture(BOOL bUICulture)
         return NULL;
     }
 
-    // Get the actual thread culture.
-    OBJECTREF pCurThreadObject = GetExposedObject();
-    _ASSERTE(pCurThreadObject!=NULL);
+    OBJECTREF pCurrentCulture;
+    if (bUICulture) {
+        // Call the Getter for the CurrentUICulture.  This will cause it to populate the field.
+        MethodDescCallSite propGet(METHOD__CULTURE_INFO__GET_CURRENT_UI_CULTURE);
+        ARG_SLOT retVal = propGet.Call_RetArgSlot(NULL);
+        pCurrentCulture = ArgSlotToObj(retVal);
```

Comment:
This gets the `CurrentUICulture` for the calling thread, not the thread represented by the `Thread` object. It didn't really work correctly before, so in terms of functionality it's just as broken as before, but a comment would be nice.